File: context-tree/tests/fixtures/languages/go/sample.go
// Package sample is a fixture for Go extraction tests.
package sample

import (
	"fmt"
	"os"
)

// GLOBAL is a package-level constant.
const GLOBAL = 10

// Value returns the sentinel CTX_SENTINEL_GODOC_3e8f marker.
//
// A second doc line so a later --doc render has more than one line.
func Value() int {
	return GLOBAL
}

// Widget is a sample struct type.
type Widget struct {
	Name string
}

// Render calls Value across symbols and touches the os package.
func (w Widget) Render() int {
	fmt.Println(w.Name)
	return Value() + os.Getpid()
}

File: context-tree/tests/fixtures/languages/rust/sample.rs
//! Crate-level fixture for Rust extraction tests.

use std::collections::HashMap;

/// A crate-level constant.
pub const GLOBAL: i32 = 10;

/// Return the sentinel CTX_SENTINEL_RUSTDOC_2d9a marker.
///
/// A second doc line so a later --doc render has more than one line.
pub fn value() -> i32 {
    GLOBAL
}

/// A sample struct type.
pub struct Widget {
    name: String,
}

impl Widget {
    /// Render calls value across symbols and touches HashMap.
    pub fn render(&self) -> i32 {
        let mut map = HashMap::new();
        map.insert(0, self.name.len());
        value() + GLOBAL
    }
}
